Update request: php83Extensions.swoole 5.0.3 → 5.1.1

- Package name: php83Extensions.swoole
- Latest released version: 5.1.1
<!-- Search your package here: https://search.nixos.org/packages?channel=unstable -->
- Current version on the unstable channel: 5.0.3
- Current version on the stable/release channel: 5.0.3
<!--
Type the name of your package and try to find an open pull request for the package
If you find an open pull request, you can review it!
There's a high chance that you'll have the new version right away while helping the community!
-->
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

<!-- If the search.nixos.org result shows no maintainers, tag the person that last updated the package. -->
@aanderse
@drupol
@etu
@Ma27
@talyz

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Version 5.1.0 fixed compatibility of this extension with PHP 8.3, so the current version 5.0.3 is unusable with PHP 8.3. It outputs these build warnings/errors

```
In file included from /private/tmp/nix-build-php-swoole-5.0.3.drv-0/source/ext-src/php_swoole.cc:16:
In file included from /private/tmp/nix-build-php-swoole-5.0.3.drv-0/source/ext-src/php_swoole_cxx.h:19:
/private/tmp/nix-build-php-swoole-5.0.3.drv-0/source/ext-src/php_swoole_private.h:927:9: error: no matching function for call to 'php_>
        php_url_encode_hash_ex(HASH_OF(zdata), formstr, NULL, 0, NULL, 0, NULL, 0, NULL, NULL, (int) PHP_QUERY_RFC1738);
        ^~~~~~~~~~~~~~~~~~~~~~
/nix/store/1pcghr2r5lv6wsszp0l6ildih94s84vw-php-8.3.0-dev/include/php/ext/standard/php_http.h:24:13: note: candidate function not viab>
PHPAPI void php_url_encode_hash_ex(HashTable *ht, smart_str *formstr,
            ^
In file included from /private/tmp/nix-build-php-swoole-5.0.3.drv-0/source/ext-src/php_swoole.cc:30:
/private/tmp/nix-build-php-swoole-5.0.3.drv-0/source/ext-src/stubs/php_swoole_ex_arginfo.h:16:35: warning: incomplete universal charac>
        ZEND_ARG_OBJ_INFO(0, ctx, Swoole\\NameResolver\\Context, 0)
                                         ^
/private/tmp/nix-build-php-swoole-5.0.3.drv-0/source/ext-src/stubs/php_swoole_ex_arginfo.h:20:34: warning: incomplete universal charac>
        ZEND_ARG_OBJ_INFO(0, ns, Swoole\\NameResolver, 0)
                                        ^
2 warnings and 1 error generated.
make: *** [Makefile:210: ext-src/php_swoole.lo] Error 1
```

--- a/pkgs/development/php-packages/swoole/default.nix
+++ b/pkgs/development/php-packages/swoole/default.nix
@@ -1,7 +1,14 @@
-{ lib, stdenv, buildPecl, php, valgrind, pcre2, fetchFromGitHub }:
+{ lib
+, stdenv
+, buildPecl
+, php
+, valgrind
+, pcre2
+, fetchFromGitHub
+ }:
 
 let
-  version = "5.0.3";
+  version = "5.1.2";
 in buildPecl {
   inherit version;
   pname = "swoole";
@@ -10,19 +17,19 @@ in buildPecl {
     owner = "swoole";
     repo = "swoole-src";
     rev = "v${version}";
-    sha256 = "sha256-xadseYMbA+llzTf9JFIitJK2iR0dN8vAjv3n9/e7FGs=";
+    hash = "sha256-WTsntvauiooj081mOoFcK6CVpnCCR/cEQtJbsOIJ/wo=";
   };
 
   buildInputs = [ pcre2 ] ++ lib.optionals (!stdenv.isDarwin) [ valgrind ];
 
-  doCheck = true;
-  checkTarget = "tests";
+  # tests require internet access
+  doCheck = false;
 
-  meta = with lib; {
+  meta = {
     changelog = "https://github.com/swoole/swoole-src/releases/tag/v${version}";
     description = "Coroutine-based concurrency library for PHP";
-    license = licenses.asl20;
-    homepage = "https://www.swoole.co.uk/";
-    maintainers = teams.php.members;
+    homepage = "https://www.swoole.com";
+    license = lib.licenses.asl20;
+    maintainers = lib.teams.php.members;
   };
 }

